presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeHiveExternalTableQueries.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.presto.hive.metastore.Column;
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.hive.HiveType.HIVE_DATE;
+import static com.facebook.presto.hive.HiveType.HIVE_DOUBLE;
+import static com.facebook.presto.hive.HiveType.HIVE_INT;
+import static com.facebook.presto.hive.HiveType.HIVE_LONG;
+import static com.facebook.presto.hive.HiveType.HIVE_STRING;
+import static com.facebook.presto.nativeworker.PrestoNativeQueryRunnerUtils.createExternalTable;
+import static org.testng.Assert.assertEquals;
+
+public abstract class AbstractTestNativeHiveExternalTableQueries
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected void createTables()
+    {
+        String schemaName = "default";
+        String tableName1 = "hive_symlink_table";
+        List<Column> tableName1Columns = ImmutableList.of(new Column("col", HIVE_LONG, Optional.empty(), Optional.empty()));
+
+        String tableName2 = "orders_symlink_table";
+        List<Column> ordersColumns = ImmutableList.of(
+                new Column("orderkey", HIVE_LONG, Optional.empty(), Optional.empty()),
+                new Column("custkey", HIVE_LONG, Optional.empty(), Optional.empty()),
+                new Column("orderstatus", HIVE_STRING, Optional.empty(), Optional.empty()),
+                new Column("totalprice", HIVE_DOUBLE, Optional.empty(), Optional.empty()),
+                new Column("orderdate", HIVE_DATE, Optional.empty(), Optional.empty()),
+                new Column("orderpriority", HIVE_STRING, Optional.empty(), Optional.empty()),
+                new Column("clerk", HIVE_STRING, Optional.empty(), Optional.empty()),
+                new Column("shippriority", HIVE_INT, Optional.empty(), Optional.empty()),
+                new Column("comment", HIVE_STRING, Optional.empty(), Optional.empty()));
+
+        String tableName3 = "lineitem_symlink_table";
```

Comment:
Since these table names are in your control, is there a chance that we can simply call these tables lineitem, order, etc and do the tpch queries on them. 

If you are able to generate the data files from the tpch catalog tables as well. that would make for a really good e2e test imo.

wdyt ?

Reply:
Thanks for reviewing the PR @aditi-pandit 
I think running all tpch queries sounds like a good idea. But all the tpch tables have to be external and data for all the tables has to be maintained under [test/resources](https://github.com/prestodb/presto/tree/master/presto-native-execution/src/test/resources)
Would it be Ok?

Reply:
Actually, no creating data for symlink on the fly wouldn't be possible direclty, since Presto doesn't support creating symlink tables. So I had to maintain the data in [test/resources](https://github.com/prestodb/presto/tree/master/presto-native-execution/src/test/resources) and create external tables of type symlink.
This is why currently I have data for 3 tables under [test/resources](https://github.com/prestodb/presto/tree/master/presto-native-execution/src/test/resources) and I am creating an external table on top of those for tests.

Reply:
We should be able to CTAS parquet tables from the tpch catalog and convert it to a symlink tables on the fly using a bash script. The symlink table contains the manifest folder with a manifest file listing the parquet files.
We can invoke the bash script from Java.

Reply:
I think converting a directory of parquet files to a Presto symlink table is pretty useful for developers.
It is similar to using a Presto file metastore.

Reply:
Can we also get this ordersColumn from the tpch connector API?

Reply:
@aditi-pandit I have added all the tpch tables and included to run all TPCH queries in the tests.

@majetideepak I have included a Java Utility named `HiveSymlinkManifestGeneratorUtils` to handle creation and cleanup of symlink related data.

> Can we also get this ordersColumns from the tpch connector API?

Could you please point me to some example for this? As of now I have added metadata columns for all tpch tables.

Reply:
> Could you please point me to some example for this?

We can do this in Prestissimo/Velox by directly using the TPCH classes.
For Java, I do see the tests invoking TpchMetadata to get these details? Can you check?
presto-tpch/src/test/java/com/facebook/presto/tpch/TestTpchMetadata.java

Reply:
> Java Utility named HiveSymlinkManifestGeneratorUtils

This looks great! We can now add/scale Delta tests in the future without having to add any data/metadata to the repo.

Reply:
I have updated the PR. Please review